### PR TITLE
feat(rust): various updates to udp rendezvous service

### PR DESCRIPTION
**Pull request by neil2468**
This is part of the ongoing protoype work on UDP NAT Hole Punching (#3507).
I'm submitting PRs bit by bit to try and keep things mangeable and avoid overwhelming.
Thanks!

## Proposed changes

- Add example `rendezvous_server` which is useful for manual testing.
- Add public `UdpRendezvousService` and make private the messages and worker of the Rendezvous service.
- As parts of Rendezvous service are private, move tests.
- Modify Rendezvous service to store the return route to a remote node instead of just a single UDP `Address`.
- Add `Ping` and `Pong` messages to Rendezvous service so remote nodes can test if service is reachable.

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs [hunk 1]
```diff
@@ -1,24 +1,29 @@
-use std::net::SocketAddr;
-
-use ockam_core::{Message, Result};
+use ockam_core::{Message, Result, Route};
 use serde::{Deserialize, Serialize};
 
 /// Request type for UDP Hole Punching Rendezvous service
 #[derive(Serialize, Deserialize, Debug, Message)]
 pub enum RendezvousRequest {
+    /// Update serivce's internal table with the
```
Comment (etorreborre) [inline implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs]: s/serivce/service

File: implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+    /// return route of the sending node.
     Update {
         /// Name of sending node
         node_name: String,
     },
+    /// Query serivce's internal table for the public
```
Comment (etorreborre) [inline implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs]: s/serivce/service

**Review by etorreborre — DISMISSED** [review]
Nice progress!

File: implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs (hunk 1 above)
Comment (neil2468) [reply implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs]: Nice catch! Done.

File: implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs (hunk 2 above)
Comment (neil2468) [reply implementations/rust/ockam/ockam_transport_udp/src/rendezvous_service/messages.rs]: Nice catch. Done.

**Review by etorreborre — APPROVED** [review]

**Pull request merged**